### DISCUSSION TITLE
Support proxy or pac files.

### DISCUSSION THREAD

**Discussion by Monirzadeh** [go-shiori/shiori#707] [Feature requests]
it is good if shiori can get PAC file to use proxy for external connection.
unfortunately some person in world don't have free access to internet and they are under heavy censorship for access to the internet or need specific rule for accesses specific site.
support proxy can help them a lot (more that you think)

any body have better idea instead of impediment that in application level that work without problem?

**Comment by fmartingr** [comment]
Have you tried using the `HTTP_PROXY` and `HTTPS_PROXY` variables? Go should respect those when making requests.

**Reply by Monirzadeh** [reply]
unfortunately some website should not pass through proxy.
i am not test that yet but maybe i should mix that with something like proxychain (it is platform specific) to bypass them.
pass all the traffic from proxy chain make some problem.

**Reply by fmartingr** [reply]
yeah, I checked a bit about PAC files and they do a bit more than just setting up the proxies, but I still think this should be on a system level rather than on the application level. 

Sorry I can't provide more help with this since I'm not very familiar with these kind of things.

**Reply by Monirzadeh** [reply]
don't worry. i work on that and test your suggestion for find a good solution.
unfortunately i should know about this stuff.

**Reply by mrchang2017** [reply]
Hello. It didn't work on docker container while `HTTP_PROXY`, `HTTPS_PROXY` existing. Maybe we should provide some variable to the HTTP client function.

**Reply by Monirzadeh** [reply]
i test that on local machine not docker yet (proxy is on docker it self that listen on specific port) and work fine. the way i done unfortunately is not exactly PAC file but similar.
what is the problem exactly? 
why you need set`HTTP_PROXY` `HTTPS_PROXY` for other propose?
shiori work correctly?
your proxy exclude local range IP?
what type of proxy you use for `HTTP_PROXY`,` HTTPS_PROXY` do you set socks?
can you provide some link that you have problem with it?
do you get something like this in log?
`error downloading boorkmark: error downloading url: Get "your url": net/http: TLS handshake timeout`

i am not close this because not test that myself on docker yet and not deep enough.

**Reply by mrchang2017** [reply]
In my Shiori Container, I set `HTTP_PROXY`, `HTTPS_PROXY`, `http_proxy`, `https_proxy` to **"http://hostname.local:7890"**, which is a http proxy. But I saw no fetching action from Shiori in Proxy log file.

Also in Shiori Log file:
```
2023/09/14 15:07:10 error downloading boorkmark: error downloading url: Get "{URL}": dial tcp 173.252.*.*:443: i/o timeout
```

**Reply by mrchang2017** [reply]
Hello, Buddy; I found that in the Shiori Container, `NO_PROXY`, `no_proxy` environment variables were both set to "*". I deleted it and the proxy worked.
